### PR TITLE
Add module for end 2 end tests for verify the the generated client co…

### PR DESCRIPTION
**Pull request by zettelmj**
…mplies with the spec and is run able

### CODE REVIEW

**Comment by zettelmj** [comment]
@cjbooms I would propose to add acceptance/end2end tests like this.
As I am working on the jdk client this comes handy to make sure that both versions of the generated client code work the same way. For the new client I would like to run the same set of tests with both clients.

**Comment by cjbooms** [comment]
> @cjbooms I would propose to add acceptance/end2end tests like this.
> As I am working on the jdk client this comes handy to make sure that both versions of the generated client code work the same way. For the new client I would like to run the same set of tests with both clients.

Yep, fine with me. Nice that it's a separate module

File: end2end-tests/src/test/kotlin/com/cjbooms/fabrikt/clients/okio3/Okio3Test.kt [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+            urlPath like "/example-path-2/304"
+        } returns {
+            statusCode = 304
+        }
+
+        val result = assertThrows<ApiException> {
+            examplePath2Client.getExamplePath2PathParam("304", 10)
+        }
+
+        assertThat(result.message).isEqualTo("[304]: ")
```
Comment (zettelmj) [inline end2end-tests/src/test/kotlin/com/cjbooms/fabrikt/clients/okio3/Okio3Test.kt:220]: @cjbooms This looks dodgy. There seems to be no handling for the 3xx cases.
Messages are either `successful -> 200..299`, `bad request -> 400..499` or `server error -> 500..599`. 
Everything 3xx falls through into the else case in `http-util.kt.hbs` line 46.

Comment (cjbooms) [reply end2end-tests/src/test/kotlin/com/cjbooms/fabrikt/clients/okio3/Okio3Test.kt:220]: I don't think anyone at Zalando have used the API client for anything other than tests, so probably proper 3X handling never came up.

**Comment by zettelmj** [comment]
First part of the coverage done

**Comment by zettelmj** [comment]
#257

**Comment by cjbooms** [comment]
👍

**Review by cjbooms — APPROVED** [review]

**Pull request merged**